### PR TITLE
Add tidiers for `fixest` models

### LINKED ISSUE #785: feature request: supporting `fixtest` objects
``` r
library(fixest)

gravity_results <- feglm(Euros ~ log(dist_km) | Origin + Destination + Product + Year, trade)

gravity_results_ols <- feols(log(Euros) ~ log(dist_km) | Origin + Destination + Product + Year, trade)

gravity_results_negbin <- fenegbin(Euros ~ log(dist_km) | Origin + Destination + Product + Year, trade)

class(gravity_results)
#> [1] "fixest"

class(gravity_results_ols)
#> [1] "fixest"

class(gravity_results_negbin)
#> [1] "fixest"

summary(gravity_results)
#> GLM estimation, family = poisson, Dep. Var.: Euros
#> Observations: 38,325 
#> Fixed-effects: Origin: 15,  Destination: 15,  Product: 20,  Year: 10
#> Standard-errors: Clustered (Origin) 
#>              Estimate Std. Error z value  Pr(>|z|)    
#> log(dist_km)  -1.5279   0.115699 -13.206 < 2.2e-16 ***
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> Log-likelihood: -7.025e+11   Adj. Pseudo-R2: 0.76403 
#>            BIC:  1.405e+12     Squared Cor.: 0.61202

summary(gravity_results_ols)
#> OLS estimation, Dep. Var.: log(Euros)
#> Observations: 38,325 
#> Fixed-effects: Origin: 15,  Destination: 15,  Product: 20,  Year: 10
#> Standard-errors: Clustered (Origin) 
#>              Estimate Std. Error z value  Pr(>|z|)    
#> log(dist_km)  -2.1699   0.154312 -14.062 < 2.2e-16 ***
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> Log-likelihood: -75,682.54   Adj. R2: 0.70514 
#>                            R2-Within: 0.21932

summary(gravity_results_negbin)
#> Negative Binomial ML estimation, Dep. Var.: Euros
#> Observations: 38,325 
#> Fixed-effects: Origin: 15,  Destination: 15,  Product: 20,  Year: 10
#> Standard-errors: Clustered (Origin) 
#>              Estimate Std. Error z value  Pr(>|z|)    
#> log(dist_km)  -1.7108   0.166378 -10.283 < 2.2e-16 ***
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
#> Over-dispersion parameter: theta = 0.5487735  
#> Log-likelihood:  -646,586.98   Adj. Pseudo-R2: 0.03464 
#>            BIC: 1,294,419.32     Squared Cor.: 0.4376
```

<sup>Created on 2019-11-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  ctype    English_United States.1252  
#>  tz       Europe/Berlin               
#>  date     2019-11-19                  
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version    date       lib source                        
#>  assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.6.0)                
#>  backports     1.1.5      2019-10-02 [1] CRAN (R 3.6.1)                
#>  callr         3.3.2      2019-09-22 [1] CRAN (R 3.6.1)                
#>  cli           1.1.0      2019-03-19 [1] CRAN (R 3.6.0)                
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.1)                
#>  desc          1.2.0      2019-11-11 [1] Github (r-lib/desc@61205f6)   
#>  devtools      2.2.1      2019-09-24 [1] CRAN (R 3.6.1)                
#>  digest        0.6.22     2019-10-21 [1] CRAN (R 3.6.1)                
#>  ellipsis      0.3.0      2019-09-20 [1] CRAN (R 3.6.1)                
#>  evaluate      0.14       2019-05-28 [1] CRAN (R 3.6.0)                
#>  fixest      * 0.2.0      2019-11-19 [1] CRAN (R 3.6.1)                
#>  Formula       1.2-3      2018-05-03 [1] CRAN (R 3.5.0)                
#>  fs            1.3.1      2019-05-06 [1] CRAN (R 3.6.0)                
#>  glue          1.3.1      2019-03-12 [1] CRAN (R 3.6.0)                
#>  highr         0.8        2019-03-20 [1] CRAN (R 3.6.0)                
#>  htmltools     0.4.0      2019-10-04 [1] CRAN (R 3.6.1)                
#>  knitr         1.26       2019-11-12 [1] CRAN (R 3.6.1)                
#>  lattice       0.20-38    2018-11-04 [2] CRAN (R 3.6.1)                
#>  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.1)                
#>  MASS          7.3-51.4   2019-03-31 [1] CRAN (R 3.6.0)                
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.6.0)                
#>  nlme          3.1-140    2019-05-12 [2] CRAN (R 3.6.1)                
#>  numDeriv      2016.8-1.1 2019-06-06 [1] CRAN (R 3.6.0)                
#>  pkgbuild      1.0.6      2019-10-09 [1] CRAN (R 3.6.1)                
#>  pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.6.0)                
#>  prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.1)                
#>  processx      3.4.1      2019-07-18 [1] CRAN (R 3.6.1)                
#>  ps            1.3.0      2018-12-21 [1] CRAN (R 3.6.0)                
#>  R6            2.4.1      2019-11-12 [1] CRAN (R 3.6.1)                
#>  Rcpp          1.0.3      2019-11-08 [1] CRAN (R 3.6.1)                
#>  remotes       2.1.0      2019-06-24 [1] CRAN (R 3.6.0)                
#>  rlang         0.4.1      2019-10-24 [1] Github (r-lib/rlang@30feeac)  
#>  rmarkdown     1.17       2019-11-13 [1] CRAN (R 3.6.1)                
#>  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.1)                
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.6.0)                
#>  stringi       1.4.3      2019-03-12 [1] CRAN (R 3.6.0)                
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.6.0)                
#>  testthat      2.3.0      2019-11-05 [1] CRAN (R 3.6.1)                
#>  usethis       1.5.1.9000 2019-11-13 [1] Github (r-lib/usethis@c5f1e7f)
#>  withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.1)                
#>  xfun          0.11       2019-11-12 [1] CRAN (R 3.6.1)                
#>  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.1)                
#> 
#> [1] C:/Users/inp099/Documents/R/win-library/3.6
#> [2] C:/Program Files/R/R-3.6.1/library
```

</details>

### PR DESCRIPTION
**Pull request by karldw**
This PR adds support for models in the _fixest_ package estimated by the functions `fixest::feglm()`, `fixest::fenegbin()`, `fixest::feNmlm()`, `fixest::femlm()`, `fixest::feols()`, and `fixest::fepois()`. Support and test coverage is highest for `feols` models.

Let me know if you want me to make changes. i expect this build to fail for (at least) two reasons:
- I use a glance column called `within.r.squared`  (depends on https://github.com/alexpghayes/modeltests/pull/23)
- I rely on not-yet-on-CRAN bug fixes for the fixest package.

Closes #785

### CODE REVIEW

File: R/fixest-tidiers.R [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+#' @family fixest tidiers
+#' @seealso [tidy()], [fixest::feglm()], [fixest::fenegbin()],
+#' [fixest::feNmlm()], [fixest::femlm()], [fixest::feols()], [fixest::fepois()]
+tidy.fixest <- function(x, conf.int = FALSE, conf.level = 0.95, ...) {
+  has_dots <- !missing(...)
+  # These two options are necessary because the user might specify
+  # broom(x, cluster="something") or
+  # broom(summary(x, cluster="something"))
+  # and calling summary() again will overwrite the arguments in the second case.
+  # Note that class(summary(x)) == class(x) == "fixest".
```
Comment (alexpghayes) [inline R/fixest-tidiers.R]: Big oooof. This is rough to deal with. Okay, so are you checking if the input object is `summary(fixest)` or `fixest` based on the other arguments the user gives this function? That seems brittle. I would prefer logic along the lines of

```
tidy.fixest {
  # figure out if we have have a `fixest` or a `summary(fixest`) object
  # call an appropriate helper
}

tidy_summary_fixest <- function(...) {
  # do stuff
}

tidy_fixest <- function(...) {
  # do other stuff
}
```

and then some error if the passed arguments and the passed `fixest` are incompatible.

File: R/fixest-tidiers.R [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      CI <- stats::confint(x, level = conf.level, ...)
+      # Bind to rest of tibble
+      colnames(CI) <- c("conf.low", "conf.high")
+      ret <- cbind(ret, unrowname(CI))
+    } else {
+      # It's possible we were passed a summary of a fixest object. If no `...`
+      # arguments were specified, then we want to use the std error that `x`
+      # already has. That is, we don't want to call confint() because
+      # doing so throws away whatever arguments were provided to summary().
+      ci_fact <- abs(qnorm((1 - conf.level) / 2))
```
Comment (alexpghayes) [inline R/fixest-tidiers.R]: This assumptions infinity degrees of freedom, correct? Is there somewhere we can get this information? If not, options include:

- Giving an error with a suggestion for the user on how to get a correct CI
- Giving the asymptotic CI, and clearly documenting this

File: R/fixest-tidiers.R [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  as_tibble(ret)
+}
+
+#' @templateVar class fixest
+#' @template title_desc_augment
+#'
+#' @inherit tidy.fixest params examples
+#' @template param_data
+#' @template param_newdata
+#' @param type.predict Passed to [`predict.fixest`][fixest::predict.fixest()]
```
Comment (alexpghayes) [inline R/fixest-tidiers.R]: What other options are there? They should be listed here.

File: R/fixest-tidiers.R [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+#' data, so you must provide it manually.
+#'
+#' augment.fixest only works for [fixest::feols()], [fixest::feglm()], and
+#' [fixest::femlm()] models.
+#'
+#' @export
+#' @family fixest tidiers
+#' @seealso [augment()], [fixest::feglm()], [fixest::femlm()], [fixest::feols()]
+augment.fixest <- function(x, data = NULL, newdata = NULL, type.predict="link", ...) {
+  if (!x$method %in% c("feols", "feglm", "femlm")) {
```
Comment (alexpghayes) [inline R/fixest-tidiers.R]: GAH if only `fixest` had real classes!

File: R/fixest-tidiers.R [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+#' @note The columns of the result depend on the type of model estimated.
+#'
+#' @return A one-row [tibble::tibble()] with exactly one row and columns:
+#'
+#' - logLik The log-likelihood of the model. [stats::logLik()] may be a useful
+#'   reference.
+#' - AIC Akaike's Information Criterion for the model.
+#' - BIC Bayesian Information Criterion for the model.
+#' - nobs Number of observations used.
+#' - sigma (`feols` models only)
```
Comment (alexpghayes) [inline R/fixest-tidiers.R]: What do you think always returning the same columns, but the value is just `NA` for non `feols` objects? Does that feel less magical?

File: tests/testthat/test-fixest.R [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  )
+
+  check_augment_function(
+    aug = augment.fixest,
+    model = fit2,
+    data = df,
+    newdata = df,
+    strict = FALSE
+  )
+
```
Comment (alexpghayes) [inline tests/testthat/test-fixest.R]: These tests are great, thank you!

File: R/fixest-tidiers.R (hunk 2 above)
Comment (karldw) [reply R/fixest-tidiers.R]: Hmm, I'm just borrowing from the `confint.fixest` code: https://github.com/lrberge/fixest/blob/acfb6877633d7fbf75a7571c15238e857c73c9ee/R/MiscFuns.R#L6197-L6202

File: R/fixest-tidiers.R (hunk 5 above)
Comment (karldw) [reply R/fixest-tidiers.R]: I don't have strong feelings here. Would you also want to make the `pseudo.r.squared` column's value `NA` for `feols` objects? 
Another option is glossing over the difference between R<sup>2</sup> and pseudo-R<sup>2</sup>, and just returning whichever is appropriate for the model.

File: R/fixest-tidiers.R (hunk 2 above)
Comment (alexpghayes) [reply R/fixest-tidiers.R]: Let's add a note to the documentation stating that CIs assume infinite degrees of freedom then.

File: R/fixest-tidiers.R (hunk 5 above)
Comment (alexpghayes) [reply R/fixest-tidiers.R]: > Would you also want to make the pseudo.r.squared column's value NA for feols objects?

Yes, let's do this.

**Comment by alexpghayes** [comment]
Not quite sure what https://travis-ci.org/tidymodels/broom/builds/625107899#L1724 means but it looks like there's some issue with dependencies

**Comment by karldw** [comment]
Ah, sorry, it's because the fixest package hasn't created a new release with some changes my code depends on. I'll have travis run the tests again when fixest issues another CRAN release.

**Comment by karldw** [comment]
The tests failed because the `joineRML` package isn't available. It's been [removed](https://cran.r-project.org/web/packages/joineRML/index.html) from CRAN. 
What do you want to do?

**Comment by karldw** [comment]
`joineRML` is back and Travis tests pass! Appveyor is struggling with the dependencies, but I don't think that's related to code I changed.

**Comment by karldw** [comment]
Sorry to generate additional alerting noise. This isn't at all urgent. I requested a review to signal that I don't have any more planned updates, but whenever you have the free time to review this would be great!

**Comment by karldw** [comment]
The changes I just pushed aim to keep this PR up to date with broom. It looks like the tests are failing on other tidiers, with some warnings about my spelling mistakes. I'm happy to push changes to fix my spelling mistakes, just let me know what's helpful.

**Comment by simonpcouch** [comment]
Thanks for all of your patience with this PR! Merged.🙂

**Pull request merged**